lib/states/statesInMemServer.js

```diff
@@ -1,1032 +1,267 @@
+            this.unsubscribeLogForClient(socket, id, callback);
+        });
+        socket.on('getSession', (id, callback) => {
+            this.getSession(id, callback);
+        });
+        socket.on('setSession', (id, expire, obj, callback) => {
+            this.setSession(id, expire, obj, callback);
+        });
+        socket.on('destroySession', (id, callback) => {
+            this.destroySession(id, callback);
+        });
+        /*
+        socket.on('getConfig', function (id, callback) {
+            this.getConfig.apply(this, arguments);
+        });
+        socket.on('getConfigKeys', function (pattern, callback, dontModify) {
+            this.getConfigKeys.apply(this, arguments);
+        });
+        socket.on('getConfigs', function (keys, callback, dontModify) {
+            this.getConfigs.apply(this, arguments);
+        });
+        socket.on('setConfig', function (id, obj, callback) {
+            this.setConfig.apply(this, arguments);
+        });
+        socket.on('delConfig', function (id, callback) {
+            this.delConfig.apply(this, arguments);
+        });
+        socket.on('subscribeConfig', function (pattern, callback) {
+            arguments.unshift(socket);
+            this.subscribeConfig.apply(this, arguments);
+        });
+        socket.on('unsubscribeConfig', function (pattern, callback) {
+            arguments.unshift(socket);
+            this.unsubscribeConfig.apply(this, arguments);
+        });
+        */
+        socket.on('setBinaryState', (id, data, callback) => {
+            this.setBinaryState(id, data, callback);
+        });
+        socket.on('getBinaryState', (id, callback) => {
+            this.getBinaryState(id, callback);
+        });
+        socket.on('delBinaryState', (id, callback) => {
+            this.delBinaryState(id, callback);
+        });
+        socket.on('error', err => this.log.error(this.namespace + ' socket.io states: ' + err));
+    }
+
+    getClients() {
+        return this.server.io.sockets.connected;
+    }
+
+    // Destructor of the class. Called by shutting down.
+    destroy() {
+        super.destroy();
+
+        if (this.server.io) {
+            if (this.server.io.sockets && this.server.io.sockets.connected) {
+                for (const s in this.server.io.sockets.connected) {
+                    if (this.server.io.sockets.connected.hasOwnProperty(s)) {
+                        delete this.server.io.sockets.connected[s];
+                    }
+                }
+            }
+            try {
+                this.server.io.close();
+            } catch (e) {
+                console.log(e.message);
+            }
+        }
+    }
+
+    initSocket(socket) {
+        if (this.settings.auth) {
+            const user = null;
+            this.socketEvents(socket, user);
+        } else {
+            this.socketEvents(socket);
+        }
+    }
+
+    _initWebServer(settings, server) {
```

Comment:
Isn't this duplicated between the two MemServer classes? Could be partially extracted into another module.

Reply:
Yes it is the same ... but we do not have "multi inheritence" :-)

yes we could extract into a "socketioTools.js" or such and put init and destroy and maybe also publishToClients there ...  is it really worth it?

Reply:
I'm just talking about initWebServer. The implementation is pretty specific and does not really need `this`, except for logging and 1 or 2 assignments. Its just something one might forget when updating one class, but not the other.

Reply:
The balance question here is for me "a new file just because one method" ... ;-) But I see your point

Reply:
One method used in two separate locations, yes that warrants a new file in my opinion. I'd rather have 200 modules with a clear intent than having to search where a method is also duplicated to :)

Reply:
Ok, lets do the following:  we revisit that again when I do the redis protocol changes :-)